### PR TITLE
feat(checkbox-group): support options

### PR DESCRIPTION
**Pull request by chouchouji**
## Checklist

List of tasks you have already done and plan to do.

- [x] Fix linting errors
- [x] Tests have been added / updated (or snapshots)

## Change information

Describe your modifications here.

## Issues

The issues you want to close, formatted as close #1.

## Related Links

Links related to this pr.

### CODE REVIEW

File: packages/varlet-ui/types/checkboxGroup.d.ts [hunk 1]
```diff
@@ -5,17 +5,24 @@ import {
...
+  disabled?: boolean
+  value: any
+}
+
 export { CheckboxGroupDirection }
 
 export interface CheckboxGroupProps extends BasicAttributes {
   modelValue?: any[]
   max?: string | number
+  options?: Array<string | number | CheckboxOption>
```
[the remark this answers is not in the document]
Comment (chouchouji) [reply packages/varlet-ui/types/checkboxGroup.d.ts:25]: label 我支持了插槽和函数，是不是value必填好些，没有传label，就默认为value

File: packages/varlet-ui/src/checkbox-group/CheckboxGroup.vue [hunk 2]
```diff
@@ -1,30 +1,56 @@
...
     const max = computed(() => props.max)
     const checkedCount = computed(() => props.modelValue.length)
+    const checkboxOptions = computed(() => {
+      const options = isArray(props.options) ? props.options : []
+      return options.map((option) => {
+        if (isNumber(option) || isString(option)) {
+          return {
+            label: option,
+            value: option,
+          } as CheckboxOption
```
[the remark this answers is not in the document]
Comment (chouchouji) [reply packages/varlet-ui/src/checkbox-group/CheckboxGroup.vue]: 这个我准备option的内容确定完了之后在shared里面加一个

File: packages/varlet-ui/src/checkbox-group/CheckboxGroup.vue [hunk 3]
```diff
@@ -1,30 +1,33 @@
 <template>
   <div :class="n('wrap')">
     <div :class="classes(n(), n(`--${direction}`))">
-      <slot />
+      <checkbox-group-option v-if="checkboxGroupOptions.length" :options="checkboxGroupOptions" />
```
Comment (haoziqaq) [inline packages/varlet-ui/src/checkbox-group/CheckboxGroup.vue:4]: 这里作为 Option 组件，应该把 v-for 逻辑放在外层来。

Comment (chouchouji) [reply packages/varlet-ui/src/checkbox-group/CheckboxGroup.vue:4]: 哦，忘了这个是option组件，不是options

File: packages/varlet-ui/src/checkbox-group/CheckboxGroupOption.tsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    label: [String, Function, Object] as PropType<string | VNode | CheckboxGroupLabelRender>,
+    value: [String, Number, Boolean, Object, Array] as PropType<any>,
+    disabled: Boolean,
+    checked: Boolean,
+  },
+  setup(props) {
+    return () => (
+      <Checkbox checkedValue={props.value} disabled={props.disabled}>
+        {isFunction(props.label)
+          ? props.label({ label: props.label, disabled: props.disabled, value: props.value }, props.checked)
```
[the remark this answers is not in the document]
Comment (chouchouji) [reply packages/varlet-ui/src/checkbox-group/CheckboxGroupOption.tsx]: 这里props.option我用了！，耗子哥看下有没有问题

**Review by haoziqaq — APPROVED** [review]

**Pull request merged**